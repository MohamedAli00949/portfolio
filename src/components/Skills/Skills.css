/* start skills section*/
section#Skills h2 {
  font-size: 40px;
  margin-block: 20px;
  color: #09c6f9;
  text-shadow: 2px 2px 5px #00000059;
  text-align: center;
}

ul.skills-container {
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  gap: 20px;
}

.skills-container .skill-card {
  position: relative;
  width: 220px;
  height: 250px;
  background: #265585;
  box-shadow: 0 0 3px 3px #489ff8c4;
  border-radius: 3%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.skills-container .skill-card .circle,
.skills-container .skill-card .circle svg {
  width: 150px;
  height: 150px;
  position: relative;
}

.skill-card .circle svg {
  transform: rotate(270deg);
}

.skills-container .skill-card .circle {
  background: #ffffff80;
  border-radius: 50%;
}

.skills-container .skill-card .circle svg circle {
  width: 100%;
  height: 100%;
  fill: transparent;
  stroke-width: 10;
  stroke: #ffffffec;
  transform: translate(5px, 5px);
}

.skills-container .skill-card .circle svg circle:nth-child(2) {
  stroke: #09c6f9;
  stroke-width: 5;
  stroke-dasharray: 440;
}

.skill-card .dot {
  position: absolute;
  inset: 5px;
  z-index: 10;
  transform: rotate(calc(3.6deg * var(--present)));
}

.dot::before {
  content: "";
  position: absolute;
  width: 15px;
  height: 15px;
  top: -7.5px;
  left: 50%;
  transform: translate(-50%);
  border-radius: 50%;
  background: #09c6f9;
  box-shadow: 0 0 10px #09c6f9, 0 0 30px #09c6f9;
}

.skill-details {
  position: absolute;
  display: flex;
  align-items: center;
  justify-content: center;
  inset: 0;
  flex-direction: column-reverse;
  color: white;
  font-weight: 700;
}

.skill-details p {
  margin-block: 0px;
}

.skill-details p:nth-child(2) {
  font-size: 35px;
  text-shadow: 0 0 6px #2c2e3d;
  margin-bottom: 0;
}

/* end skills section*/