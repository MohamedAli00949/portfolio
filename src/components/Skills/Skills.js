import { Container } from "@mui/material";
import React from "react";

const Skills = () => {
  const skillsData = [
    {
      name: "HTML",
      rate: "95",
    },
    {
      name: "CSS",
      rate: "85",
    },
    {
      name: "JavaScript",
      rate: "90",
    },
    {
      name: "React.js",
      rate: "75",
    },
    {
      name: "Redux.js",
      rate: "70",
    },
    {
      name: "Node.js",
      rate: "70",
    },
    {
      name: "Exprees.js",
      rate: "70",
    },
    {
      name: "MongooDB",
      rate: "70",
    },
    {
      name: "Mongoose.js",
      rate: "75",
    },
    {
      name: "SQL",
      rate: "60",
    },
    {
      name: "C++",
      rate: "60",
    },
    {
      name: "Virsion control",
      rate: "80",
    },
  ];
  return (
    <section
      id="Skills"
      style={{
        background: "rgb(37, 63, 80)",
        color: "white",
        padding: "132px 0px",
      }}
    >
      <Container fixed>
        <div>
          <h2>Skills</h2>
          <div>
            <ul className="skills-container">
              {skillsData.map((skill, index) => (
                <li key={index} className="skill-card">
                  <div className="circle" style={{'--present': skill.rate}}>
                    <div className="dot"></div>
                    <svg>
                      <circle cx="70" cy="70" r="70"></circle>
                      <circle cx="70" cy="70" r="70" style={{strokeDashoffset: `calc(440 - (440 * ${skill.rate} /100))`}}></circle>
                    </svg>
                    <div className="skill-details">
                      <p style={{color: "rgb(42 42 56)"}}>{skill.name}</p>
                      <p>{skill.rate}%</p>
                    </div>
                  </div>
                </li>
              ))}
            </ul>
          </div>
        </div>
      </Container>
    </section>
  );
};

export default Skills;
