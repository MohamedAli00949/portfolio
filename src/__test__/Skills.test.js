import React, { Suspense } from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import { act } from "react-dom/test-utils";

import Skills from "../components/Skills/Skills";

let container = null;
beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    container.setAttribute("id", "containeer")
    document.body.appendChild(container);
});

afterEach(() => {
    ReactDOM.unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("render footer section", async () => {
    await act(async () => {
        await render(<Suspense><Skills /></Suspense>)
    })

    container = await document.body;

    expect(container.querySelector("section").getAttribute("id")).toBe("Skills");
    expect(container.querySelector("h2").textContent).toBe("Skills");

    expect(container.querySelectorAll("ul li").length).toBe(12);
    
    expect(container.querySelector("ul li .circle"));
    expect(container.querySelector("ul li .circle .dot"));
    expect(container.querySelectorAll("ul li .circle svg circle").length).toBe(2*12);

    expect(container.querySelector("ul li .skill-details p").textContent).toBe("HTML");
    expect(container.querySelectorAll("ul li .skill-details p")[1].textContent).toBe("95%");

    expect(container).toMatchSnapshot();
})