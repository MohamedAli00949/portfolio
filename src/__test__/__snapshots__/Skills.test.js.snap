// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render footer section 1`] = `
<body>
  <div
    id="containeer"
  />
  <div>
    <section
      id="Skills"
      style="background: rgb(37, 63, 80); color: white; padding: 132px 0px;"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-fixed css-1ss2ei0-MuiContainer-root"
      >
        <div>
          <h2>
            Skills
          </h2>
          <div>
            <ul
              class="skills-container"
            >
              <li
                class="skill-card"
              >
                <div
                  class="circle"
                  style="--present: 95;"
                >
                  <div
                    class="dot"
                  />
                  <svg>
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                    />
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                      style="stroke-dashoffset: calc(440 - (440 * 95 /100));"
                    />
                  </svg>
                  <div
                    class="skill-details"
                  >
                    <p>
                      HTML
                    </p>
                    <p>
                      95
                      %
                    </p>
                  </div>
                </div>
              </li>
              <li
                class="skill-card"
              >
                <div
                  class="circle"
                  style="--present: 85;"
                >
                  <div
                    class="dot"
                  />
                  <svg>
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                    />
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                      style="stroke-dashoffset: calc(440 - (440 * 85 /100));"
                    />
                  </svg>
                  <div
                    class="skill-details"
                  >
                    <p>
                      CSS
                    </p>
                    <p>
                      85
                      %
                    </p>
                  </div>
                </div>
              </li>
              <li
                class="skill-card"
              >
                <div
                  class="circle"
                  style="--present: 90;"
                >
                  <div
                    class="dot"
                  />
                  <svg>
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                    />
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                      style="stroke-dashoffset: calc(440 - (440 * 90 /100));"
                    />
                  </svg>
                  <div
                    class="skill-details"
                  >
                    <p>
                      JavaScript
                    </p>
                    <p>
                      90
                      %
                    </p>
                  </div>
                </div>
              </li>
              <li
                class="skill-card"
              >
                <div
                  class="circle"
                  style="--present: 75;"
                >
                  <div
                    class="dot"
                  />
                  <svg>
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                    />
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                      style="stroke-dashoffset: calc(440 - (440 * 75 /100));"
                    />
                  </svg>
                  <div
                    class="skill-details"
                  >
                    <p>
                      React.js
                    </p>
                    <p>
                      75
                      %
                    </p>
                  </div>
                </div>
              </li>
              <li
                class="skill-card"
              >
                <div
                  class="circle"
                  style="--present: 70;"
                >
                  <div
                    class="dot"
                  />
                  <svg>
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                    />
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                      style="stroke-dashoffset: calc(440 - (440 * 70 /100));"
                    />
                  </svg>
                  <div
                    class="skill-details"
                  >
                    <p>
                      Redux.js
                    </p>
                    <p>
                      70
                      %
                    </p>
                  </div>
                </div>
              </li>
              <li
                class="skill-card"
              >
                <div
                  class="circle"
                  style="--present: 70;"
                >
                  <div
                    class="dot"
                  />
                  <svg>
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                    />
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                      style="stroke-dashoffset: calc(440 - (440 * 70 /100));"
                    />
                  </svg>
                  <div
                    class="skill-details"
                  >
                    <p>
                      Node.js
                    </p>
                    <p>
                      70
                      %
                    </p>
                  </div>
                </div>
              </li>
              <li
                class="skill-card"
              >
                <div
                  class="circle"
                  style="--present: 70;"
                >
                  <div
                    class="dot"
                  />
                  <svg>
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                    />
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                      style="stroke-dashoffset: calc(440 - (440 * 70 /100));"
                    />
                  </svg>
                  <div
                    class="skill-details"
                  >
                    <p>
                      Exprees.js
                    </p>
                    <p>
                      70
                      %
                    </p>
                  </div>
                </div>
              </li>
              <li
                class="skill-card"
              >
                <div
                  class="circle"
                  style="--present: 70;"
                >
                  <div
                    class="dot"
                  />
                  <svg>
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                    />
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                      style="stroke-dashoffset: calc(440 - (440 * 70 /100));"
                    />
                  </svg>
                  <div
                    class="skill-details"
                  >
                    <p>
                      MongooDB
                    </p>
                    <p>
                      70
                      %
                    </p>
                  </div>
                </div>
              </li>
              <li
                class="skill-card"
              >
                <div
                  class="circle"
                  style="--present: 75;"
                >
                  <div
                    class="dot"
                  />
                  <svg>
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                    />
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                      style="stroke-dashoffset: calc(440 - (440 * 75 /100));"
                    />
                  </svg>
                  <div
                    class="skill-details"
                  >
                    <p>
                      Mongoose.js
                    </p>
                    <p>
                      75
                      %
                    </p>
                  </div>
                </div>
              </li>
              <li
                class="skill-card"
              >
                <div
                  class="circle"
                  style="--present: 60;"
                >
                  <div
                    class="dot"
                  />
                  <svg>
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                    />
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                      style="stroke-dashoffset: calc(440 - (440 * 60 /100));"
                    />
                  </svg>
                  <div
                    class="skill-details"
                  >
                    <p>
                      SQL
                    </p>
                    <p>
                      60
                      %
                    </p>
                  </div>
                </div>
              </li>
              <li
                class="skill-card"
              >
                <div
                  class="circle"
                  style="--present: 60;"
                >
                  <div
                    class="dot"
                  />
                  <svg>
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                    />
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                      style="stroke-dashoffset: calc(440 - (440 * 60 /100));"
                    />
                  </svg>
                  <div
                    class="skill-details"
                  >
                    <p>
                      C++
                    </p>
                    <p>
                      60
                      %
                    </p>
                  </div>
                </div>
              </li>
              <li
                class="skill-card"
              >
                <div
                  class="circle"
                  style="--present: 80;"
                >
                  <div
                    class="dot"
                  />
                  <svg>
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                    />
                    <circle
                      cx="70"
                      cy="70"
                      r="70"
                      style="stroke-dashoffset: calc(440 - (440 * 80 /100));"
                    />
                  </svg>
                  <div
                    class="skill-details"
                  >
                    <p>
                      Virsion control
                    </p>
                    <p>
                      80
                      %
                    </p>
                  </div>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </section>
  </div>
</body>
`;
